extensions/amp-animation/0.1/web-animation-service.js

```diff
@@ -33,14 +33,15 @@ export class WebAnimationService {
   }
 
   /**
+   * @param {!Document|!ShadowRoot|!Element} root
```

Comment:
@dvoytenko You mentioned offline that the `rootNode` concept for `amp-animation` is meant for shadow roots.

It seems it's only consumed by `CssContext` for `querySelectors`. Could you provide more context on the issues caused by using this to scope to the `<amp-story-page>`?

Reply:
TBH, I've lately been trying to be much more explicit in difference between root nodes and any other scoping elements. We've had too many bugs when either one or the other was used incorrectly. In your case specifically, the `rootNode` is always available and only sometimes you might want to rescope it. So the rootNode-vs-scoping element is purely the matter of the internal implementation code here, and I don't really see the reason to add to the implementation ambiguity if we only save on a single argument in 1-2 places?

To stress: not all of these is needed for the first pull request :)

But to answer your question, here are some things that a rootNode is used for (possibly not a complete list):
 * Look for elements: `getElementById()` and `querySelectorAll()`.
 * DOM readiness. E.g. before you can `querySelectorAll` you need to make sure the document is complete.
 * Access to the stylesheets. As in `document.styleSheets` or `shadowRoot.styleSheets`.
 * Calculation of root-based properties, such as root font size (`rem`).

Based on this, scoping rules are:
 * `querySelectorAll` -> scoping element.
 * `getElementById` -> scoping element, as `querySelector`. And we have to keep in mind that it's slightly more complicated than a simple `'#' + id`.
 * DOM Readiness. The most elegant way for us here is to bypass this entirely and make it the responsibility of the story page. E.g. we assume that the relevant DOM is fully ready when story page builds the animation.
 * Stylesheets: rootNode.styleSheets.filter(s => scopingElement.contains(s.styleNode)).

Additional for us that may or may not need to be connected to the scoping element directly:
 * getViewportSize -> scoping element's boudingClientRect. Or maybe manually passing `width/height`. 
 
Stretching it (maybe a lot and for later/never):
 * window.onresize: this relates to both `getViewportSize` and animation restarting. We can probably continue using it. But if the story page can resize w/o window resize - we need to propagate a signal to here.
 * We recalc `vw`/`vh` and the viewport size. To be fully consistent that we should also recalc `@media` queries. We don't parse them currently and simply call `Window.matchMedia`, so it'd seem like a lot of work to support them.
 * We also have root font size, but I think it's such a mess in CSS today, that we're best steering clear of it.

Reply:
Thank you, this is very thorough!

Removing scoping in favor of a later PR.